r/src/array_to_vector.cpp

```diff
@@ -294,22 +294,25 @@ struct Converter_String : public Converter {
                                                 array->offset(), n);
       for (int i = 0; i < n; i++, null_reader.Next()) {
```

Comment:
Please wrap these loops in a call to `unwind_protect`:
```suggestion
      cpp11::unwind_protect([&] {
        for (int i = 0; i < n; i++, null_reader.Next()) {
```